### DISCUSSION TITLE
[feature request] reverse dependency list

### DISCUSSION THREAD

**Discussion by jepessen** [microsoft/vcpkg#17741] [Ideas]
I think that this feature request is most for developer of vcpkg than for the user.

Example:

I want to change a portfile for updating a library, in order to solve issue with another one:

for example, now I'd like to work with `tiff` library because there's a issue with `osg` that doesn't link `tiff` correctly. In order to solve this I need to update `tiff` library instead of `osg`.

But changing tiff portfile probably can create problems to other libraries that depends on it, and I'd like to know what they are in order to try to build all of them.

the command that I want should be something like

```
vcpkg reverse-dependency-list tiff:x64-windows
```

and it shows all libraries in vcpks that needs tiff in order to be built. The result should be something like

```
   osg[core]:x64-windows
* anotherlibrary[core]:x64-windows
etc
```
`*` prefix represents libraries that needs to be installed. With this I can check what libraries depends from the one I'm working on and when I update it test it with all of them in order to be sure that I'm not broking any working dependencies.

**Comment by ccorn** [comment]
Just commenting: MacPorts `port` has subcommands (and associated selectors)
* deps (packages that this depends on)
* rdeps (recursive deps)
* dependents (installed packages that depend on this)
* rdependents (recursive dependents)

You seem to want something like `dependents`.

**Comment by ccorn** [comment]
I'd have to caution that even MacPorts `dependents` only considers installed packages. If you want to query all dependents from the ports database, the following command might help, assuming you use git-bash and `cd` into your `vcpkg` clone directory:
~~~
grep -ri --include=CONTROL "Depends:.*\<tiff\>" ports
~~~
and if there are no false positives, you could reduce the output:
~~~
grep -rli --include=CONTROL "Depends:.*\<tiff\>" ports | awk -F/ '{print $2}'
~~~

**Comment by jepessen** [comment]
That's a nice solution, but I think that a native one will be better. Even if I manage the repository on windows from a git bash, I run it from Powershell, and I think that most of windows users will use powershell or cmd, that don't contain unix commands.

**Comment by Neumann-A** [comment]
try `findstr` on windows

**Comment by PhoebeHui** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen.

**Comment by MikeGitb** [comment]
Please reopen - I'd also be interested in that feature.

**Comment by JackBoosY** [comment]
Please use command `./vcpkg depend-info PORT_NAME --sort=reverse`.

**Reply by dg0yt** [reply]
> Please use command `./vcpkg depend-info PORT_NAME --sort=reverse`.

This changes the output order, but it doesn't list ports which *depend on*  PORT_NAME.

**Reply by JackBoosY** [reply]
Oh, I know what he meant.
